code-challenges.test.js

```diff
@@ -23,13 +23,31 @@ var people = [
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe('stringFormatter', () => {
+  it('capitalizes the names and interpolates properly', () => {
+    expect(stringFormatter(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthur Dent is a radio employee."]
 
+/*
+ * Loop through the array with .map(), using an anonymous function with arguments of destructured key value pairs, i.e. array.map(({name, occupation}) => {return sentence})
+ * return the output of that map
+ * inside map capitalize each word in name with name.split(' ') and a loop
+ * interpolate using both variables
+*/
 
 // b) Create the function that makes the test pass.
 
+const stringFormatter = people => {
+  return people.map(({name, occupation}) => {
+    let capital = name.split(' ').map(str => str[0].toUpperCase() + str.substr(1)).join(' ') // adds the first letter capitalized to the substring of the word starting at the 2nd letter
+    return `${capital} is a ${occupation}.`
+  })
+}
```

Comment:
(daaaangit I can't believe I forgot about destructuring in the argument when I did this...)
KELEN, great work man! Love the pseudo code and what a great solution you've got here!